.sync/github_templates/contributing/CONTRIBUTING.md

```diff
@@ -46,6 +46,49 @@ If you cannot find an existing issue that describes your bug or feature, create
 Please continue to follow your request after it is submitted to assist with any additional information that might be
 requested.
 
+### Pull Request Best Practices
+
+Pull requests for UEFI code can become large and difficult to review due to the large number of build and
+configuration files. To aid maintainers in reviewing your code, we suggest adhering to the following guidelines:
+
+1. Do keep code reviews single purpose; don't add more than one feature at a time.
+2. Do fix bugs independently of adding features.
+3. Do provide documentation and unit tests.
+4. Do introduce code in digestible amounts.
+   * If the contribution logically be broken up into separate pull requests that independently build and function
+     successfully, do use multiple pull requests.
+
+#### Code Categories
+
+To keep code digestible, you may consider breaking large pull requests into three categories of commits within the pull request.
+
+1. **Interfaces**: .h, .inf, .dec, documentation
+2. **Implementation**: .c, unit tests, unit test build file; unit tests should build and run at this point
+3. **Integration/Build**: .dec, .dsc, .fdf, integration tests; code added to platform and affects downstream consumers

```

Comment:
should Integration/build mention automation files such as yaml?

Reply:
I think yaml is worth mentioning. For the average contributor, that would more likely mean something like the package CI yaml file where they need to adjust spell check exceptions.

Maybe say "configuration files (.yaml)" to generically state those types of files should be handled there.

Reply:
Though in some cases (spell check exception) the configuration files could apply to any of the 3 steps?

Reply:
The main reason for suggesting (3) is that CI plugins run against the package CI config file only run during build/integration. As a logical contribution unit, it would be necessary for the content added for that step of the process.

YAML can also mean things like pipeline files, but those are very minimal in repos at this point (due to reuse from mu_devops) and the files that are in the repos are synced from mU_devops. I'd prefer not to bring that up, it increases the verbosity which will lose people's interest and those that need to change those files know this or can point someone to the right files in the rare off chance they're not aware how those work.

Reply:
I'm also okay for those changes to be added in the commits that introduce the spelling error (or whatever else). I don't really have a strong opinion since those have little to no impact on my review pace.

Reply:
Done.